tableau_utilities/tableau_server/tableau_server.py

```diff
@@ -193,14 +203,7 @@ def get_datasources(self):
             URI GET /api/api-version/sites/site-id/datasources
         Returns: All datasources in the site
         """
-        url = f"{self.url}/datasources?fields=_default_" \
-              f",favoritesTotal" \
-              f",databaseName" \
-              f",connectedWorkbooksCount" \
-              f",hasAlert" \
-              f",hasExtracts" \
-              f",isPublished" \
-              f",serverName"
+        url = f"{self.url}/datasources"
```

Comment:
This was erroring for me with the fields limit so I just removed it. I get a full response with all the information now

Reply:
Interesting, were you using a specific api version? Wonder why I didnt get the error before when I tested this 🤔 

But if you're getting all of the fields I specified here, I suppose it doesn't really matter

Reply:
So looking at this again I was passing in version 2.8 as that was the version in the code I copied to get started here.
```
    parser.add_argument('--api_version', help='Tableau API version', default='2.8')
```

That said here's a sample response with the fields you had specified in bold. Looks like everything is here.  So if you're cool making this change it seems like we might as well get all the info here and then we could do exception handling somewhere else as needed

Datasource(id='5c5b68a9-d03d-4101', 
name='Daily Account Status', 
created_at=datetime.datetime(2022, 12, 9, 20, 13, 31), 
updated_at=datetime.datetime(2022, 12, 9, 23, 4, 4), 
project_id='24b3ac9e-56f6-4cb3-aa32', 
project_name='Adhoc Requests', 
owner_id='1c27acef-60b1-42b4-b274', 
tags=None, 
description=None, 
content_url='DailyAccountStatus', 
type='snowflake', 
encrypt_extracts=None, 
**has_alert=None,** 
**has_extracts=None,** 
is_certified=False, 
**is_published=None,** 
use_remote_query_agent=None, 
**database_name=None,** 
**server_name=None,** 
**favorites_total=None,** 
**connected_workbooks_count=None,** 
webpage_url=None, 
certification_note=None)

Reply:
Can you try running it again with the original code and the latest api version?
The `Datasource` dataclass is going to provide all of those attributes by default, set to `None`.
So the fact that this is what you are seeing makes me think those things are not actually returned by the API.

Reply:
@JustinGrilli  your original code works fine with the latest version of the server of `3.17`

I don't understand what the reason for having these params is but I'm happy to revert this. Why do you want to specify these?

Reply:
Just because they can be useful for people, and I think they were provided when we were using `tableauserverclient`, so I wanted to keep them for consistency.

But yeah sounds good. Rather not restrict fields just to be compliant with old api versions.

Reply:
Ok. sounds like you want me to take out those params.  Doing that and resolving this.

Reply:
haha no I was thinking the opposite.
Since those values wont be provided without specifying them in the URL request, we should live them there. This may mean we have to use a later version of the API, but I think that preferable anyway.